src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3060,21 +3060,25 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               setError(fun)
             }
           } else {
-            fun.body match {
+            if (pt.typeSymbol == PartialFunctionClass) {
```

Comment:
I'd have to get out my computer glasses, I find big if/else blocks so hard to read compared to matches. Can the 
`} else { if (PFC) { block } else doTF }` 
be written 
`} else if (PFC) { block } else doTF`? shifting the block left.

Apologies in advance if already discussed.

Reply:
Good idea.  Done.